opencog/atoms/base/Atom.h

```diff
@@ -190,6 +195,16 @@ class Atom
         return classserver().isA(at, t);
     }
 
+    /** Returns the outgoing set.
+     *
+     * @return The outgoing set.
+     */
+    virtual inline const HandleSeq& getOutgoingSet() const
+    {
+        static HandleSeq no_atoms;
+        return no_atoms;
+    }
```

Comment:
All I am saying is that it *is* a programming error to ask for the name of an atom that is not actually a node.  By throwing an exception, the programmer who makes that mistake is alerted of their mistake immediately. It's got nothing to do with python -- its an error in any language binding ...C++ or guile or haskell or python.

As to performance, I was under the impression that the C++ cast to a NodePtr was "free" and was done at compile time. By contrast, a virtual method call costs run-time CPU cycles.  Are you saying I mis-understood the NodeCast?  I'm not sure if I ever looked at the assembly that gets generated, so maybe I'm wrong on this.  Any, that would be a good reason to NOT use virtual calls.